app/main.py

```diff
@@ -1,67 +1,212 @@
+import hashlib
 import json
+from datetime import datetime, timedelta, timezone
+from functools import lru_cache
+from pathlib import Path
 from typing import MutableMapping
 from uuid import UUID
 
 import uvicorn
 import yaml
 from fastapi import FastAPI, Query, HTTPException
+from pydantic import BaseModel
 from sdc.crypto.jwe_helper import JWEHelper
 from sdc.crypto.key_store import KeyStore
 
-KEY_PURPOSE_SDS = "supplementary_data"
-
 app = FastAPI()
 
 with open("dev-keys.yml", encoding="UTF-8") as keys_file:
     keys = KeyStore(yaml.safe_load(keys_file))
 
+KEY_PURPOSE_SDS = "supplementary_data"
+MOCK_DATA_ROOT_PATH = Path(__file__).parent.parent / "mock_data"
+
+# period_id to match with Launcher
+PERIOD_ID = "201605"
+
+# Currently only have a single schema version
+SCHEMA_VERSION = "v1.0.0"
+
+# Currently only have 1 reporting unit example for each schema version
+TOTAL_REPORTING_UNITS = 1
+
+FORM_TYPES = ["001"]
+
+# Hardcoded paths for now - update if any changes are made to https://github.com/ONSdigital/sds-schema-definitions/tree/main/examples
+MOCK_DATA_PATHS_BY_SURVEY_ID = {
+    "test": ["123"],
+    "prodcom": ["014"],
+    "bres_and_brs": [
+        "221",  # BRES
+        "241",  # BRS
+    ],
+    "roofing_tiles_slate": [
+        "068",  # Roofing tiles
+        "071",  # Slate
+    ],
+    "sand_and_gravel": [
+        "066",  # Sand and & Gravel (Land Won)
+        "076",  # Sand & Gravel (Marine Dredged)
+    ],
+}
+
+
+class DatasetMetadata(BaseModel):
+    """
+    Model for SDS Metadata Response
+    copied from https://github.com/ONSdigital/sds/blob/main/src/app/models/dataset_models.py#L19
+    """
+
+    # Required fields
+    survey_id: str
+    period_id: str
+    form_types: list[str]
+    sds_published_at: str
+    total_reporting_units: int
+    schema_version: str
+    sds_dataset_version: int
+    filename: str
+    dataset_id: UUID
+
+    # Optional fields
+    title: str | None = None
+
+
+class UnitData(BaseModel):
+    """
+    Model for SDS Unit Data Response
+    copied from https://github.com/ONSdigital/sds/blob/main/src/app/models/dataset_models.py#L53
+    """
+
+    dataset_id: UUID
+    survey_id: str
+    period_id: str
+    form_types: list[str]
+    schema_version: str
+    data: str
+
 
 @app.get("/v1/unit_data")
-def get_sds_data(
-    dataset_id: UUID, identifier: str = Query(min_length=1)
-) -> MutableMapping:
+def get_unit_data(dataset_id: UUID, identifier: str = Query(min_length=1)) -> UnitData:
```

Comment:
I've tried using dot notation e.g. `dataset_id: UnitData.dataset_id`, but pydantic gives an error `AttributeError: dataset_id` when I try to start up the service. I'm not certain but it might be trying to access the class member value rather than the type of that member? There might be a way around this but I'm not sure?

Reply:
Guessing there was no way around this?

Reply:
I've tried a few different things but I can't seem to get around it. It keeps giving me the `AttributeError: dataset_id` on startup when I try changing it to `dataset_id: UnitData.dataset_id`